text/0032-dust-protection/0032-dust-protection.md

```diff
@@ -0,0 +1,114 @@
+                                                        <td><b>Name</b></td>
+                                                        <td><b>Type</b></td>
+                                                        <td><b>Description</b></td>
+                                                    </tr>
+                                                    <tr>
+                                                        <td>Address Type</td>
+                                                        <td>uint8</td>
+                                                        <td>
+                                                            Set to <strong>value 1</strong> to denote an <i>Ed25519 Address</i>.
+                                                        </td>
+                                                    </tr>
+                                                    <tr>
+                                                        <td>Address</td>
+                                                        <td>Array&lt;byte&gt;[32]</td>
+                                                        <td>The raw bytes of the Ed25519 address which is a BLAKE2b-256 hash of the Ed25519 public key.</td>
+                                                    </tr>
+                                                </table>
+                                            </details>
+                                        </td>
+                                    </tr>
+                                    <tr>
+                                        <td>Amount</td>
+                                        <td>uint64</td>
+                                        <td>The amount of tokens to deposit with this <i>SigLockedSingleOutput</i> output.</td>
+                                    </tr>
+                                </table>
+
+
+### The new protocol dust rule
+
+Let `A` be the address that should hold the dust outputs' balances. Let `M` be the sum of all SigLockedDustAllowanceOutputs' values on `A`.So the total allowed dust outputs on `A` is (`M`/1 MIOTA) * 100 rounded down to the nearest integer. So 100 outputs for 1 MIOTA deposited.
+
+The SigLockedDustAllowanceOutput must be greater or equal to 1MI.
+
+If while processing a milestone a new dust UTXO is created but there is not a sufficient amount already locked in SigLockedDustAllowanceOutputs then the encompassing message should be marked as `ignored`.
+
+If while processing a milestone an unspent SigLockedDustAllowanceOutput is consumed and as a result we have a violation on the number of allowed dust outputs then the encompassing message should be marked as `ignored`.
+
+The checks should happen after the entire message was processed. Messages that were previously applied by the current milestone are taken into account.
+
+
+# Drawbacks
+
+1. We can't have an address that will ever hold less than 1 MI. It will be a bit hard on testnet faucets but still workable.
```

Comment:
```suggestion
- There can no longer be addresses holding less than 1 Mi.
```
I don't think testnet faucets would be the main issue here, this could matter for quite a few projects, so maybe just leave this example out.